### PR TITLE
install.sh:set aio conf during installation

### PR DESCRIPTION
**Pull request by yaronkaikov**
This is a follow up change to https://github.com/scylladb/scylla/pull/8512

Let's add aio conf file during scylla installation process and make sure
we also remove this file when uninstall Scyll

### CODE REVIEW

**Comment by syuu1228** [comment]
Could you explain bit more why we need this PR?
Seems like #8512 already fixes previous problem (#8587)

**Review by penberg — APPROVED** [review]
lgtm, @avikivity, @syuu1228 please review

**Review by syuu1228 — CHANGES_REQUESTED** [review]

**Comment by avikivity** [comment]
We've spent a lot of effort on aio-max-nr:

588a065304d8d292f99672d613afa11797c89059
53c7600da8c21664628ed0241851e4ee71025690
d0297c599afadd808f00f88ac4b9f55631cb543a
950dbdb4660104a448fab5852ce8ed40e539e1b0

How about we go back to static configuration, and make it large enough to work with 500 cpus.

**Comment by syuu1228** [comment]
Hmm, I do not know what is the issue of current dynamic configuration of aio-max-nr setting, but if it is bad to configure it during scylla_io_setup, we might configure it at scylla_post_install.sh and handle package manager as ghost file, just like /etc/systemd/system/scylla-server.service.d/capabilities.conf.
 
Even this solution doesn't work, I agree to back to static configuration, but probably better to print warning message on scylla_setup to tell users 'manual configuration of aio-max-nr setting requires for ncpus > 500 environment.'.

**Comment by yaronkaikov** [comment]
@syuu1228 I think that the reason we configure this aio-max-nr during `scylla_io_setup` is mainly for the machine-images 
I don't see how we can remove it

**Comment by avikivity** [comment]
> Hmm, I do not know what is the issue of current dynamic configuration of aio-max-nr setting, but if it is bad to configure it during scylla_io_setup,

I don't know if there is a specific problem here. I do know that we keep seeing problems and fixes and then new problems, and there is no real need for dynamic configuration.

> we might configure it at scylla_post_install.sh and handle package manager as ghost file, just like /etc/systemd/system/scylla-server.service.d/capabilities.conf.
> 
> Even this solution doesn't work, I agree to back to static configuration, but probably better to print warning message on scylla_setup to tell users 'manual configuration of aio-max-nr setting requires for ncpus > 500 environment.'.

Seastar only supports 256 shards, so that's okay.

**Comment by syuu1228** [comment]
> > Hmm, I do not know what is the issue of current dynamic configuration of aio-max-nr setting, but if it is bad to configure it during scylla_io_setup,
> 
> I don't know if there is a specific problem here. I do know that we keep seeing problems and fixes and then new problems, and there is no real need for dynamic configuration.
> 
> > we might configure it at scylla_post_install.sh and handle package manager as ghost file, just like /etc/systemd/system/scylla-server.service.d/capabilities.conf.
> > Even this solution doesn't work, I agree to back to static configuration, but probably better to print warning message on scylla_setup to tell users 'manual configuration of aio-max-nr setting requires for ncpus > 500 environment.'.
> 
> Seastar only supports 256 shards, so that's okay.

Ah okay, then static configuration idea seems reasonable (without warning message that I suggested).

**Comment by yaronkaikov** [comment]
@syuu1228 @avikivity @penberg Can you please review again? 
I have set the value to be static as @avikivity Suggested and tested it on both AMI and centos7 google instance

**Comment by avikivity** [comment]
@yaronkaikov by static I meant going back to shipped files instead of creating them during installation. See 950dbdb4660104a448fab5852ce8ed40e539e1b0.

**Comment by yaronkaikov** [comment]
@avikivity This is what i did, didn't i? where do you see i am generating the file during installation ?

**Comment by avikivity** [comment]
```
    with open(aio_conf_file, mode) as file:
            for line in file:
                        aio_conf_value = re.findall(r"[=]?\d*\.\d+|\d+", line)
```

**Comment by yaronkaikov** [comment]
> ```
>     with open(aio_conf_file, mode) as file:
>             for line in file:
>                         aio_conf_value = re.findall(r"[=]?\d*\.\d+|\d+", line)
> ```

Ok, i don't generate the file, just make sure that `/proc/sys/fs/aio-max-nr` value match to what we defined.

But when thinking about it, maybe it's not needed any more, so may also drop it from `scylla_io_setup` and `scylla_prepare`

**Comment by yaronkaikov** [comment]
@avikivity Removed the non-relevant code on a new commit, can you please review again ?

File: dist/common/scripts/scylla_util.py [hunk 1]
```diff
@@ -102,26 +102,15 @@ def scyllabindir():
...
-        with open(aio_conf_file, mode) as f:
-            f.write(aio_max_nr_value + '\n')
-        with open('/proc/sys/fs/aio-max-nr', mode) as f:
-            f.write(str(get_required_aio_slots()))
+    with open(aio_conf_file, mode) as file:
+        for line in file:
+            aio_conf_value = re.findall(r"[=]?\d*\.\d+|\d+", line)
+            aio_value = int((', '.join(map(str, aio_conf_value))))
+            with open('/proc/sys/fs/aio-max-nr', mode) as f:
+                f.write(str(aio_value))
```
Comment (avikivity) [inline dist/common/scripts/scylla_util.py]: I still don't see why we have this.

Comment (yaronkaikov) [reply dist/common/scripts/scylla_util.py]: We don't , it's already removed (marked as Outdated)

**Comment by yaronkaikov** [comment]
@avikivity ping

**Comment by yaronkaikov** [comment]
@avikivity Any other comments i should fix?

**Comment by yaronkaikov** [comment]
@penberg @avikivity - please review

**Comment by penberg** [comment]
@yaronkaikov Please just squash the two commits together, and I will merge.

**Comment by yaronkaikov** [comment]
@penberg Done

**Review by syuu1228 — APPROVED** [review]
LGTM.

**Pull request closed without merging**